src/dataflow/src/source/mod.rs

```diff
@@ -438,9 +447,9 @@ impl Codec for AssignedTimestamp {
         buf.extend(&self.0.to_le_bytes())
     }
 
-    fn decode<'a>(buf: &'a [u8]) -> Result<Self, String> {
+    fn decode<'a>(buf: &'a [u8]) -> Result<Self, CodecError> {
         Ok(AssignedTimestamp(u64::from_le_bytes(
-            <[u8; 8]>::try_from(buf).map_err(|err| err.to_string())?,
+            <[u8; 8]>::try_from(buf).map_err(|err| CodecError::from(err.to_string()))?,
```

Comment:
Same here for avoiding `map_err` with a `impl From<std::array::TryFromSliceError> for CodecError` implementation